From: Andreas Metzler <ametzler@debian.org>
Date: Fri, 13 May 2022 23:45:47 +0800
Subject: Version filename of locale data (gnutls30.mo instead of

gnutls.mo) This is necessary to make e.g. libgnutls26 and libgnutls28
co-installable.
Last-Update: 2020-01-18
---
 configure.ac     | 3 +++
 lib/global.c     | 2 +-
 lib/str.h        | 6 +++---
 libdane/errors.c | 4 ++--
 po/Makevars      | 2 +-
 5 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5057536..99df3ee 100644
--- a/configure.ac
+++ b/configure.ac
@@ -391,6 +391,9 @@ dnl Try the hooks.m4
 LIBGNUTLS_HOOKS
 LIBGNUTLS_EXTRA_HOOKS
 
+AC_DEFINE_UNQUOTED([GNUTLSDOMAIN], ["${PACKAGE}${DLL_VERSION}"],
+   [base filename for gettext message catalogue])
+
 AC_ARG_ENABLE(tests,
   AS_HELP_STRING([--disable-tests], [don't compile or run any tests]),
     enable_tests=$enableval, enable_tests=$enable_tools)
diff --git a/lib/global.c b/lib/global.c
index 7e5ef3d..8b57674 100644
--- a/lib/global.c
+++ b/lib/global.c
@@ -280,7 +280,7 @@ static int _gnutls_global_init(unsigned constructor)
 	}
 
 #ifdef HAVE_DCGETTEXT
-	bindtextdomain(PACKAGE, LOCALEDIR);
+	bindtextdomain(GNUTLSDOMAIN, LOCALEDIR);
 #endif
 
 	e = secure_getenv("GNUTLS_BUFFER_RECLAIM");
diff --git a/lib/str.h b/lib/str.h
index 4e305c4..5721262 100644
--- a/lib/str.h
+++ b/lib/str.h
@@ -33,9 +33,9 @@
 #include "attribute.h"
 
 #ifdef HAVE_DCGETTEXT
-#include "gettext.h"
-#define _(String) dgettext(PACKAGE, String)
-#define N_(String) gettext_noop(String)
+# include "gettext.h"
+# define _(String) dgettext (GNUTLSDOMAIN, String)
+# define N_(String) gettext_noop (String)
 #else
 #define _(String) String
 #define N_(String) String
diff --git a/libdane/errors.c b/libdane/errors.c
index 522d6f0..5d5eb93 100644
--- a/libdane/errors.c
+++ b/libdane/errors.c
@@ -25,8 +25,8 @@
 
 /* I18n of error codes. */
 #include "gettext.h"
-#define _(String) dgettext(PACKAGE, String)
-#define N_(String) gettext_noop(String)
+#define _(String) dgettext (GNUTLSDOMAIN, String)
+#define N_(String) gettext_noop (String)
 
 #define ERROR_ENTRY(desc, name) { desc, #name, name }
 
diff --git a/po/Makevars b/po/Makevars
index 81c12f7..1f11984 100644
--- a/po/Makevars
+++ b/po/Makevars
@@ -5,7 +5,7 @@
 # unlimited permission to use, copy, distribute, and modify it.
 
 # Usually the message domain is the same as the package name.
-DOMAIN = $(PACKAGE)
+DOMAIN = $(PACKAGE)30
 
 # These two variables depend on the location of this directory.
 subdir = po
